txserver/handler.go

```diff
@@ -47,15 +48,47 @@ func commit_buy(ctx *context.Context, command *Command) ([]byte, error) {
 }
 
 func cancel_buy(ctx *context.Context, command *Command) ([]byte, error) {
-	return []byte{}, nil
+	account, err := find_account(ctx, command.Username)
+	if err != nil {
+		return []byte{}, fmt.Errorf("Failed to cancel buy for %s, error: %s", command.Username, err.Error())
+	}
+
+	account.RecentBuy.Timestamp = 0
+	account.RecentBuy.Amount = 0
+	account.RecentSell.stock = ""
+
+	update := bson.M{"$set": bson.D{primitive.E{Key: "recentSell", Value: account.RecentSell}}}
```

Comment:
we need to use this because our updateOne() with setUpsert = true
![image](https://user-images.githubusercontent.com/39280329/156065030-ff5f8035-ee60-433b-93d1-56768aa4e15c.png)